cdap-app-fabric/src/main/java/io/cdap/cdap/internal/credential/store/CredentialIdentityStore.java

```diff
@@ -38,14 +42,26 @@
 import java.util.Spliterators;
 import java.util.stream.Collectors;
 import java.util.stream.StreamSupport;
+import javax.inject.Inject;
+import javax.inject.Named;
 
 /**
  * Storage for credential identities.
  */
 public class CredentialIdentityStore {
 
+  private static final byte[] CREDENTIAL_IDENTITY_STORE_AD = CredentialIdentityStore.class
```

Comment:
If we refactor and changed the class name later, will it break when trying to decrypt data that was encrypted using the old class name as the ad?

Reply:
I think it is safer to actually store the ad being used for encryption in the structured table as well so that we can use it for decryption.

Reply:
Storing the AD in the table removes the protection against an attacker who has write access to the table; in particular, they could lift a value encrypted with a different AD from another table and write it to the current table, and it will decrypt properly.

I think it's probably better to hardcode this as a string value with a note that changing the value is backwards incompatible.

Reply:
Because we are using the same key to encrypt/decrypt data in different tables?

Reply:
Yup